WordPress/Classes/ViewRelated/Post/PostSettings/Views/Excerpt/PostSettingsExcerptEditor.swift

```diff
@@ -0,0 +1,223 @@
+import SwiftUI
+import WordPressUI
+import WordPressShared
+import DesignSystem
+
+struct PostSettingsExcerptEditor: View {
+    let postContent: String
+
+    @Binding var text: String
+
+    @State private var wordCount = 0
+    @State private var isAnimating = false
+    @State private var textViewOpacity = 1.0
+    @State private var showUndoButton = false
+    @State private var previousText: String = ""
+    @State private var hideUndoButtonTask: Task<Void, Never>?
+
+    @FocusState private var isFocused: Bool
+
+    @ScaledMetric private var editorHeightCompact = 160
+    @ScaledMetric private var editorHeightRegular = 200
+
+    @Environment(\.horizontalSizeClass) private var horizontalSizeClass
+
+    private let placeholder = PostSettingExcerptRow.localizedPlaceholderText
+
+    var body: some View {
+        Form {
+            Section {
+                editor
+            } footer: {
+                HStack {
+                    Text(String.localizedStringWithFormat(Strings.characterCount, text.count))
+                        .foregroundColor(.secondary)
+                    Spacer()
+                    Text(String.localizedStringWithFormat(Strings.wordCount, wordCount))
+                        .foregroundColor(.secondary)
+                }
+                .font(.caption)
+            }
+        }
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+            ToolbarItem(placement: .topBarTrailing) {
+                if !postContent.isEmpty && LanguageModelHelper.isSupported {
```

Comment:
The generated result is not great when the post content is short. The result is heavily influenced by the prompt, rather than the "source content". Maybe we should only show the "Generate" button when the post content is at least a certain characters long?